### PR TITLE
Add From impl for kbucket::Key

### PR DESCRIPTION
**Pull request by izderadicka**
- From<record::Key>
- From<Vec<u8>
- this will enable to use additional types in kad.get_closest_peers as it was possible in pre 0.33 version

### CODE REVIEW

**Comment by izderadicka** [comment]
PR for issue #1898

File: protocols/kad/src/kbucket/key.rs [hunk 1]
```diff
@@ -25,6 +25,8 @@ use sha2::digest::generic_array::{GenericArray, typenum::U32};
 use std::borrow::Borrow;
 use std::hash::{Hash, Hasher};
```
Comment (mxinden) [inline protocols/kad/src/kbucket/key.rs:27]:
```suggestion
```

I don't think we use the convention of separating crate level imports and external imports with a newline.

**Review by mxinden — APPROVED** [review]
Looks good to me. Thanks for the patch.

File: protocols/kad/src/kbucket/key.rs (hunk 1 above)
Comment (izderadicka) [reply protocols/kad/src/kbucket/key.rs:27]: NP - this is how rust-analyzer plugin in VS Code added it

**Review by romanb — APPROVED** [review]

**Comment by mxinden** [comment]
For the record `libp2p-kad` `v0.27.1` [tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-kad-v0.27.1) and [published](https://crates.io/crates/libp2p-kad/0.27.1).

**Pull request merged**